pkg/outputs/parse.go

```diff
@@ -147,12 +149,21 @@ func parseLogLineWithChunks(text string, chunks map[string]*strings.Builder) (st
 	return text, nil
 }
 
-func Parse(chunks map[string]*strings.Builder, logText string) (*ParsedLine, error) {
+type ParseOptions struct {
+	// If non-zero, returns an error if an output line is too long.
+	MaxOutputLineLen int
+}
+
+func Parse(chunks map[string]*strings.Builder, logText string, opts ParseOptions) (*ParsedLine, error) {
 	outputText, err := parseLogLineWithChunks(logText, chunks)
 	if err != nil {
 		return nil, err
 	}
 
+	if opts.MaxOutputLineLen != 0 && opts.MaxOutputLineLen < len(outputText) {
```

Comment:
Minor: handle negative values as "disabled"

```suggestion
	if opts.MaxOutputLineLen > 0 && opts.MaxOutputLineLen < len(outputText) {
```